src/ddscxx/src/org/eclipse/cyclonedds/pub/qos/DataWriterQosDelegate.cpp

```diff
@@ -163,7 +170,8 @@ DataWriterQosDelegate::ddsc_qos() const
 #ifdef  OMG_DDS_OWNERSHIP_SUPPORT
     strength_    .delegate().set_c_policy(qos);
 #endif  // OMG_DDS_OWNERSHIP_SUPPORT
-    lifecycle_.delegate().set_c_policy(qos);
+    lifecycle_   .delegate().set_c_policy(qos);
+    datarepresentation_.delegate().set_c_policy(qos);
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro.

Reply:
Will do.